### DISCUSSION TITLE
Show user game history on ticket page

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAWeb#713] [Ideas]
As a developer, when I view a ticket, I want to quickly be able to tell if the user has since earned the problematic achievement and get a quick overview of what other achievements they may have earned or had trouble earning.

Suggest adding a new top-level field "Earned at" that would be "never" if not earned, or the timestamp where the user did eventually earn the achievement.

Suggest adding a game history block (like that which appears on the user profile page) just above Comments showing all achievements earned for the game with tooltips displaying when they were earned. The important information here is the timeline. It's important to see which achievements were earned before the ticket was submitted, and which were earned after.

**Comment by meleu** [comment]
@Jamiras as I'm currently dealing with ticketmanager code I think I can work on your request. Just would like to know your opinion here...

What do you think about a line saying something like this:

- `ReporterName` got this achievement at `AwardedTime`, which is **after** this report.
- `ReporterName` got this achievement at `AwardedTime`, which is **before** this report.
- `ReporterName` did not get this achievement

Also adding a link to that `gamecompare.php` page, should be "good enough" for you?

By the way, I want to add an option to sort that list on the `gamecompare.php` page by time. It will be also useful for investigating accusations of cheating.

**Comment by Jamiras** [comment]
The three bullet points would be great. 

A sorted list is better than embedding the icon view as you wouldn't need to mouse over each icon to see the timestamps. I'm not sure that the `gamecompare.php` page is the best solution for that as it also show the current user's achievements, so the sort is ambiguous. (sort by my time or his?). It's probably a reasonable short-term solution.

**Comment by meleu** [comment]
> I'm not sure that the `gamecompare.php` page is the best solution for that as it also show the current user's achievements, so the sort is ambiguous. (sort by my time or his?)

Yeah, I've just realized this when looking to the code now. But I thought on a solution, I'll tweak the `userInfo.php` to provide such list.

**EDIT**
not to self:
test this SQL query:
```sql
SELECT
    ach.ID,
    ach.Title,
    ach.Description,
    ach.Points,
    ach.TrueRatio,
    ach.BadgeName,
    ach.DisplayOrder,
    aw.Date,
    aw.HardcoreMode
FROM
    Achievements AS ach
LEFT JOIN
    Awarded AS aw ON aw.AchievementID = ach.ID
LEFT JOIN
    UserAccounts AS ua ON ua.User = aw.User
WHERE
    ach.GameID = $gameID
    AND ach.Flags = 3
ORDER BY
    aw.HardcoreMode DESC,
    aw.Date ASC,
    ach.ID ASC;
```

**Comment by Jamiras** [comment]
It looks like something is wrong with the timestamps. See http://retroachievements.org/ticketmanager.php?i=13293

![image](https://user-images.githubusercontent.com/32680403/41610102-915d0078-73a9-11e8-8956-51d07c5e4277.png)

The achievement triggered on ROM load, then the user submitted the ticket, but the timestamps suggest he submitted the ticket three minutes before he earned the achievement.

**Comment by meleu** [comment]
I can't be sure, but looks like he reseted and got the achievement again. He says "loaded the rom first time", and it makes me believe that it didn't happen on the next attempts.

On the achievements page the timestamp says 16:03, and the timestamp showed on the ticket's page is obtained from the same source (the `Awarded` table).

**Comment by kdecks** [comment]
@Jamiras any reason why this can't be closed?

**Comment by Jamiras** [comment]
The current solution only shows whether or the player has earned the achievement before or after creating the ticket, or not at all. The original ask also included some way to view the achievements the player has earned before and after writing the ticket so you can gauge if they're in the right part of the game, or having other issues. I still think that's a reasonable feature, but I'm fine breaking it out as a separate issue.

**Comment by Jamiras** [comment]
The player's achievement history for the game is now available via #945. This can be closed.

**Comment by Sanaki** [comment]
Discussions can't be closed, but perhaps we should add a "resolved" label where suitable.

**Reply by luchaos** [reply]
Alternatively, I prefer "locking" threads to mark discussions as resolved. 
It prevents necroing old discussions and makes them easily identifiable.
If a topic comes up again it can reference old ones.

**Reply by luchaos** [reply]
Seems I mistook that with how steam discussions work, nevermind.
